watertap/unit_models/tests/test_cstr.py

```diff
@@ -392,85 +359,30 @@ def test_block_triangularization(self, model):
         assert not model.fs.unit.inlet.temperature[0].fixed
         assert not model.fs.unit.inlet.pressure[0].fixed
 
-    @pytest.mark.solver
-    @pytest.mark.skipif(solver is None, reason="Solver not available")
-    @pytest.mark.component
-    def test_costing(self):
-        m = ConcreteModel()
-        m.fs = FlowsheetBlock(dynamic=False)
-
-        m.fs.props_ASM1 = ASM1ParameterBlock()
-        m.fs.ASM1_rxn_props = ASM1ReactionParameterBlock(
-            property_package=m.fs.props_ASM1
-        )
-
-        m.fs.unit = CSTR(
-            property_package=m.fs.props_ASM1, reaction_package=m.fs.ASM1_rxn_props
-        )
-
-        m.fs.unit.inlet.flow_vol[0].set_value(1.2199 * units.m**3 / units.s)
-        m.fs.unit.inlet.alkalinity[0].set_value(4.5102 * units.mole / units.m**3)
-        m.fs.unit.inlet.conc_mass_comp[0, "S_I"].set_value(
-            0.061909 * units.kg / units.m**3
-        )
-        m.fs.unit.inlet.conc_mass_comp[0, "S_S"].set_value(
-            0.012366 * units.kg / units.m**3
-        )
-        m.fs.unit.inlet.conc_mass_comp[0, "X_I"].set_value(
-            1.4258 * units.kg / units.m**3
-        )
-        m.fs.unit.inlet.conc_mass_comp[0, "X_S"].set_value(
-            0.090508 * units.kg / units.m**3
-        )
-        m.fs.unit.inlet.conc_mass_comp[0, "X_BH"].set_value(
-            2.8404 * units.kg / units.m**3
-        )
-        m.fs.unit.inlet.conc_mass_comp[0, "X_BA"].set_value(
-            0.20512 * units.kg / units.m**3
-        )
-        m.fs.unit.inlet.conc_mass_comp[0, "X_P"].set_value(
-            0.58681 * units.kg / units.m**3
-        )
-        m.fs.unit.inlet.conc_mass_comp[0, "S_O"].set_value(
-            0.00036092 * units.kg / units.m**3
-        )
-        m.fs.unit.inlet.conc_mass_comp[0, "S_NO"].set_value(
-            0.012424 * units.kg / units.m**3
-        )
-        m.fs.unit.inlet.conc_mass_comp[0, "S_NH"].set_value(
-            0.0076936 * units.kg / units.m**3
-        )
-        m.fs.unit.inlet.conc_mass_comp[0, "S_ND"].set_value(
-            0.0019068 * units.kg / units.m**3
-        )
-        m.fs.unit.inlet.conc_mass_comp[0, "X_ND"].set_value(
-            0.0053166 * units.kg / units.m**3
-        )
 
-        m.fs.unit.inlet.temperature[0].set_value(308.15 * units.K)
-        m.fs.unit.inlet.pressure[0].set_value(84790.0 * units.Pa)
+class TestCosting(UnitTestHarness):
+    def configure(self):
+        m = build_ASM1()
```

Comment:
In the cstr with injection test, you also test the solutions for `build_ASM1`. Is there a reason to not do the same here?

Reply:
In the injection CSTR, the saponification test is testing the CSTR part but requires a test with injection, which is why we test ASM1. ASM1 reaction is being tested in the reaction parameter block test and the flowsheets